pkgs/development/tools/java/jextract/default.nix

```
@@ -38,13 +38,13 @@ in
 
 stdenv.mkDerivation {
   pname = "jextract";
-  version = "unstable-2023-04-14";
+  version = "unstable-2023-09-25";
 
   src = fetchFromGitHub {
     owner = "openjdk";
     repo = "jextract";
-    rev = "cf3afe9ca71592c8ebb32f219707285dd1d5b28a";
-    hash = "sha256-8qRD1Xg39vxtFAdguD8XvkQ8u7YzFU55MhyyJozVffo=";
+    rev = "a3c3644580a5144840de341e7c1cbf7dc4de3206"; # Tweak build for JDK 21
+    hash = "sha256-0Z8IskIF9GT3Y+Pah2G+TvCVGBK6++/1ctirv8ZsGBA=";
```

Could we use the latest JDK21 commit? I tested it and looks good: 

```suggestion
    rev = "f8d229775e8260d4bebf21525b3151b9fff128ae";
    hash = "sha256-jkUCh4oSgilszvvU7RpozFATIghaA9rJRAdUIl5jTHM=";
```

---

I chose `a3c3644580a5144840de341e7c1cbf7dc4de3206` (`# Tweak build for JDK 21`) because it corresponds with the `jextract` binary released [on jdk.java.net](https://jdk.java.net/jextract/).

The hash you are suggesting (https://github.com/openjdk/jextract/commit/f8d229775e8260d4bebf21525b3151b9fff128ae) has a fix for a Windows test and a fix for one of the code samples so does not affect the `jextract` binary in any way.

There is also this commit https://github.com/openjdk/jextract/commit/8730fcf05c229d035b0db52ee6bd82622e9d03e9 (on `master`) that seems to have some actual fixes.

I'd be happy with any of those three commits. @SharzyL what do you suggest?

---

Since jextract doesn't have any releases yet, I'd go with https://github.com/openjdk/jextract/commit/8730fcf05c229d035b0db52ee6bd82622e9d03e9 since it included a fix. But that's my personal opinion.

---

If you find these new changes useful, it is totally fine to include these new changes. Just remember to update the commit message, PR title and do tests. jextract have no release tags hence we can and only can pick commits from master.

---

OK, I've updated to the current `master` of https://github.com/openjdk/jextract and tested on `aarch64-darwin` and it is working!